### PR TITLE
Feat: using kepler.gl to replace leaflet map

### LINKED ISSUE #45: integrate reactgeoda submodule + script
just leaving an issue here to remind to work on integration of reactgeoda as a submodule + write a script for build automation.

### PR DESCRIPTION
**Pull request by lixun910**
- move the react + webpack code to keplermap/ [test_kepler.zip](https://github.com/GeoDaCenter/data-and-lab/files/12186391/test_kepler.zip)
- copy and reconfig the webpack.config.js
- update the js libraries and the package.json

The target is to prebuild the react component for mapping first and the output will be in asserts/js/webglmap.js

- replace the old leaflet code in each post with the new <div id="root" >
- update the CSS style to match our site

### CODE REVIEW

File: _posts/2020-09-11-airbnb.md [hunk 1]
```diff
@@ -18,61 +18,7 @@ categories:
...
-        });
-      },
-      onEachFeature: onEachFeature
-    });
-    json.addTo(map);
-    map.fitBounds(json.getBounds());
-  });
-
-</script>
+<div id="root" data-geojson="../data/airbnb_Chicago2015.geojson" style="height: 800px"></div>
```
Comment (lixun910) [inline _posts/2020-09-11-airbnb.md]:
```suggestion
<div id="root" data-geojson="../data/airbnb_Chicago2015.geojson"></div>
```
@lixun910  height is not needed here

**Comment by lixun910** [comment]
@dpeachpeach Hi David, this is the react project to create the bundle.js 
[test_kepler.zip](https://github.com/GeoDaCenter/data-and-lab/files/12186391/test_kepler.zip)

**Comment by lixun910** [comment]
@dpeachpeach See the list of tasks above. Thanks!

> - move the react + webpack code to keplermap/ [test_kepler.zip](https://github.com/GeoDaCenter/data-and-lab/files/12186391/test_kepler.zip)
>- copy and reconfig the webpack.config.js
>- update the js libraries and the package.json
>
>The target is to prebuild the react component for mapping first and the output will be in asserts/js/webglmap.js
>
>- replace the old leaflet code in each post with the new <div id="root" >
>- update the CSS style to match our site

**Comment by lixun910** [comment]
@dpeachpeach  check out this branch `feat-webgl-map`, run `jekyll serve` and you should be able to see the kepler.gl map running in the first AirBnB data page.

**Comment by dpeachpeach** [comment]
yep! @lixun910 I'm integrating kepler gl into the other markdown files

**Comment by dpeachpeach** [comment]
Hey @lixun910 I tested everything on my end locally and it works ok. Fixed some styling issues. I couldn't get building to work locally so everything in the map itself style-wise is the same but I'm hoping i can solve that sometime after today. Let me know if everything looks good

**Comment by lixun910** [comment]
For conflicting files, you can rebase `main` branch into your current branch (see instructions [here](https://docs.github.com/en/desktop/contributing-and-collaborating-using-github-desktop/keeping-your-local-repository-in-sync-with-github/syncing-your-branch-in-github-desktop#rebasing-your-project-branch-onto-another-branch))

**Comment by lixun910** [comment]
ref: https://github.com/GeoDaCenter/reactgeoda/pull/36

**Comment by dpeachpeach** [comment]
Hey @lixun910 , after I resolve the conflict should I merge? Or should I commit the changes I made with integrating react-geoda

**Comment by lixun910** [comment]
> Hey @lixun910 , after I resolve the conflict should I merge? Or should I commit the changes I made with integrating react-geoda

@dpeachpeach Thanks David! You can solve the conflict first, then you can commit the changes integrating react-geoda. Then, merge the PR after I review it :)

**Review by dpeachpeach — APPROVED** [review]

**Pull request merged**